lld/MachO/InputFiles.cpp

```diff
@@ -411,10 +411,14 @@ void ObjFile::parseSections(ArrayRef<SectionHeader> sectionHeaders) {
       auto *isec = make<ConcatInputSection>(section, data, align);
       if (isDebugSection(isec->getFlags()) &&
           isec->getSegName() == segment_names::dwarf) {
-        // Instead of emitting DWARF sections, we emit STABS symbols to the
-        // object files that contain them. We filter them out early to avoid
-        // parsing their relocations unnecessarily.
+        // Keep debug sections in debugSections for diagnostic purposes
         debugSections.push_back(isec);
```

Comment:
it is not used for diagnostic purposes, it's used to initialize the DwarfObject so we can emit STABS properly. I actually think we can remove debugSections entirely now that we are keeping references to the DWARF InputSections in `sections`. We can have `DwarfObject::create` re-traverse `sections` in order to find the relevant ones.